db.go

```diff
@@ -298,6 +298,10 @@ func (db *DB) run() {
 			default:
 			}
 		case <-db.compactc:
+			if !db.compactionsIsEnabled() {
```

Comment:
If we consider : `db.compactionsEnabled()` -> `db.DisableCompactions()` -> `db.compact()`, would that still be a race?

Reply:
not sure what you mean , each one of these has a mutex  lock

Reply:
After successful `DisableCompactions` call, we should not be doing any compactions. So a `compact` call after disabling compactions doesn't sound right. Though we can logically assume that `compact` happened before in this case.